lib/frame.js

```diff
@@ -134,43 +165,42 @@ api.frame = (state, subjects, frame, parent, property = null) => {
       if(id in state.uniqueEmbeds[state.graph]) {
         _removeEmbed(state, id);
       }
-      state.uniqueEmbeds[state.graph][id] =
-        {parent, property};
     }
 
+    state.uniqueEmbeds[state.graph][id] = {parent, property};
+
     // push matching subject onto stack to enable circular embed checks
     state.subjectStack.push({subject, graph: state.graph});
 
     // subject is also the name of a graph
     if(id in state.graphMap) {
       let recurse = false;
       let subframe = null;
-      if(!('@graph' in frame)) {
+      if(!frame.hasOwnProperty('@graph')) {
```

Comment:
We don't really need to do `hasOwnProperty` for static strings.  Really more of an issue when it's a dynamic string that could conflict with an inherited Object property.  Probably doesn't matter, but the `in` syntax looks nicer when it's ok to use it.

Reply:
```suggestion
      if(!('@graph' in frame)) {
```